CourseTemplate/CourseTemplate/Program.cs

```diff
@@ -1,19 +1,36 @@
 using CourseTemplate;
 using CourseTemplate.Core;
+using CourseTemplate.Data;
 using CustomExceptions;
+using Microsoft.EntityFrameworkCore;
 
 var builder = WebApplication.CreateBuilder(args);
+// Access configuration
+var connectionString = builder.Configuration.GetConnectionString("DefaultConnection");
+
+// Configure your DbContext
+builder.Services.AddDbContext<CourseTemplateDbContext>(options =>
+	options.UseNpgsql(connectionString));
+builder.Services.AddScoped<CourseTemplateService>();
+builder.Services.AddScoped<ICourseTemplateRepository, CourseTemplateRepository>();
 
-builder.Services.AddSingleton<CourseTemplateService>();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 builder.Services.AddLogging();
+
+// Configure Forwarded Headers Middleware to handle reverse proxy server scenarios
+// builder.Services.Configure<ForwardedHeadersOptions>(options =>
+// {
+//     options.ForwardedHeaders = ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+// });
```

Comment:
Why commented?

Reply:
For future we can use proxy. So I left these comments in advance with an example of implementation